README.md

```diff
@@ -1,27 +1,134 @@
-This repository contains Windows agents and scripts for Google Compute Engine.
+## Windows Guest Environment for Google Compute Engine
+This repository stores the collection of Windows packages installed on Google 
+supported Compute Engine [images](https://cloud.google.com/compute/docs/images).
 
-This repository contains:
+**Table of Contents**
 
-+ Latest Windows agent source code and binaries -- For handling user account and address management.
-+ Latest version of the metadata scripts source code and binary -- For handling startup and shutdown scripts.
-+ Latest sysprep scripts -- For running sysprep on new Windows virtual machines.
-+ Latest VSS agent -- For handling VSS on snapshots.
+* [Background](#background)
+* [Agent](#Agent)
+    * [Account Setup](#account-setup)
+    * [IP Forwarding](#ip-forwarding)
+* [Instance Setup](#instance-setup)
+* [Metadata Scripts](#metadata-scripts)
+* [Network Setup](#network-setup)
+* [Packaging and Package Distribution](#packaging-andpackage-distribution)
+* [Contributing](#contributing)
+* [License](#license)
 
-## Contribute
+## Background
 
-Have a patch that will benefit this project? Awesome! Follow these steps to have it accepted.
+The Windows guest environment denotes the Google provided configuration and
+tooling inside of a [Google Compute Engine](https://cloud.google.com/compute/)
+(GCE) virtual machine. The
+[metadata server](https://cloud.google.com/compute/docs/metadata) is a
+communication channel for transferring information from a client into the guest.
+The Windows guest environment includes a set of scripts and binaries that read 
+the content of the metadata server to make a virtual machine run properly on our 
+platform.
 
-1. Please sign our [Contributor License Agreement](CONTRIB.md).
-1. Fork this Git repository and make your changes.
-1. Create a Pull Request
-1. Incorporate review feedback to your changes.
-1. Accepted!
+## Agent
 
-## License
+#### Account Setup
+
+The agent handles [creating user accounts and setting/resetting passwords](https://cloud.google.com/compute/docs/instances/windows/creating-passwords-for-windows-instances).
+
+#### IP Forwarding
+
+The agent uses IP forwarding metadata to setup or remove IP routes in the guest.
+
+*   Only IPv4 IP addresses are currently supported.
+
+## Instance Setup
+
+`instance_setup.ps1` is configured by GCE sysprep to run on VM first boot. The script 
+configures the Windows guest environment by performing the following tasks.
+
+*   Setting Hostname
+*   Running user provided 'specialize' startup script
+*   Activating Windows
+*   Seting up RDP and WinRM
+
+## Metadata Scripts
+
+Metadata scripts implement support for running user provided
+[startup scripts](https://cloud.google.com/compute/docs/startupscript) and
+[shutdown scripts](https://cloud.google.com/compute/docs/shutdownscript).
+
```

Comment:
Yeah that's fine. It is nice to actual document what the code is doing in the code docs though. Likewise for any metadata key we're using here.